### PR TITLE
feat: add filter by hardware

### LINKED ISSUE #407: Add filter by environment compatible

### PR DESCRIPTION
**Pull request by anajalvarenga**
Add filter by hardware and fix filter errors

## How to test
1. got to http://localhost:5173/tree/7678c1b2735a048774c98c2d3b6b4718762c8570?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentTreeDetailsTab=treeDetails.builds&diffFilter=%7B%7D&treeInfo=%7B%22gitBranch%22%3A%22for-kernelci%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Flinux%2Fkernel%2Fgit%2Farm64%2Flinux.git%22%2C%22treeName%22%3A%22arm64%22%2C%22commitName%22%3A%22v6.12-rc4-141-g7678c1b2735a%22%2C%22headCommitHash%22%3A%227678c1b2735a048774c98c2d3b6b4718762c8570%22%7D
2. try to apply some filters
3. got to http://localhost:5173/tree/e9001a382fa2c256229adc68d55212028b01d515?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=redhat&currentTreeDetailsTab=treeDetails.tests&diffFilter=%7B%7D&treeInfo=%7B%22gitBranch%22%3A%22master%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Fvirt%2Fkvm%2Fkvm.git%22%2C%22treeName%22%3A%22kvm%22%2C%22commitName%22%3A%22for-linus-30-ge9001a382fa2%22%2C%22headCommitHash%22%3A%22e9001a382fa2c256229adc68d55212028b01d515%22%7D
4. try to apply some test duration filters

- Closes #407 and #423

### CODE REVIEW

**Review by MarceloRobert — COMMENTED** [review]
Looks good, but some of the hardware filters seem to not be working, they are listed as tested hardware but when the filter is applied it retrieves 0 boots/tests. This makes it so that even when every hardware is selected in the filters, the return is less than using no filters.

File: dashboard/src/api/TreeDetails.tsx [hunk 1]
```diff
@@ -80,14 +80,16 @@ const mapFiltersToUrlSearchParams = (
...
+  const filterParam: { [key: string]: string[] } = {};
 
   Object.keys(filter).forEach(key => {
     const filterList = filter[key as keyof TTreeDetailsFilter];
-    filterList?.forEach(
-      value => (filterParam[`filter_${key}`] = value.toString()),
-    );
+    filterList?.forEach(value => {
+      if (!filterParam[`filter_${key}`])
+        return (filterParam[`filter_${key}`] = [value.toString()]);
```
Comment (Francisco2002) [inline dashboard/src/api/TreeDetails.tsx:90]: you don't need this `return` statement.

File: dashboard/src/api/TreeDetails.tsx [hunk 2]
```diff
@@ -80,14 +80,16 @@ const mapFiltersToUrlSearchParams = (
...
+      return filterParam[`filter_${key}`].push(value.toString());
```
Comment (Francisco2002) [inline dashboard/src/api/TreeDetails.tsx:91]: ditto about `return` statement

**Comment by anajalvarenga** [comment]
> The hardware filters looks good!! But the others filters in Build Tab were broken when I select more than one filter by field.
> 
> * Tree when I test: http://localhost:5173/tree/6c455c44f0ac22ab9f237ee520d0dad4ad1bda1e?tableFilter=%7B%22bootsTable%22%3A%22all%22%2C%22buildsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentTreeDetailsTab=treeDetails.builds&diffFilter=%7B%7D&treeInfo=%7B%22gitUrl%22%3A%22https%3A%2F%2Fandroid.googlesource.com%2Fkernel%2Fcommon%22%2C%22gitBranch%22%3A%22android-mainline%22%2C%22treeName%22%3A%22android%22%2C%22commitName%22%3A%22v6.12-rc3-6464-g6c455c44f0ac2%22%2C%22headCommitHash%22%3A%226c455c44f0ac22ab9f237ee520d0dad4ad1bda1e%22%7D
> 
> Wrong	Correct
> errado	correto
> errado	correto
> errado	correto
> errado	correto

I create a new issue for this bug #438

**Review by MarceloRobert — APPROVED** [review]
Working well!

**Review by lfjnascimento — APPROVED** [review]

**Pull request merged**